src/app_init/scaffold.cr

```diff
@@ -0,0 +1,95 @@
+module Mint
+  class Scaffold
+    APP = "#{__DIR__}/app"
+
+    getter path : Path
+
+    def self.run(name : Path | String)
+      new(Path[name]).run
+    end
+
+    def initialize(path : Path)
+      @path = path.expand
+    end
+
+    def name
+      path.basename
+    end
+
+    def run
+      directory =
+        name
+          .colorize(:light_green)
+          .mode(:bold)
+
+      terminal.puts "#{COG} Creating directory: #{directory}"
+
+      FileUtils.mkdir_p path.to_s
+      FileUtils.cd path.to_s
+
+      terminal.puts "#{COG} Writing initial files:"
+
+      touch_initial_files
+      touch_config
+
+      show_created_files(path.to_s, "  ")
+    end
+
+    private def touch_initial_files
+      FileUtils.cp_r(APP, "./")
+    end
+
+    private def touch_config
+        file_path = "mint.json"
+        FileUtils.mkdir_p File.dirname(file_path)
+        File.write(file_path, json.to_pretty_json)
+    end
+
+    private def show_created_files(path : String, tabs : String)
+      dirs = Dir.open(path).select do |child| 
+        child != "." &&
+        child != ".." &&
+        File.directory? File.join(path, child)
+      end
```

Comment:
Avoid using `Dir.open` without a matching call to `Dir#close`. In this case it's better to use `Dir.each_child`.

```suggestion
      dirs, files = %w[], %w[]

      Dir.each_child(path) do |child|
        child_path = File.join(path, child)
        case
        when File.directory?(child_path)
          dirs << child
        when File.file?(child_path)
          files << child
        end
      end
```